opal/corelib/runtime.js

```diff
@@ -183,7 +183,9 @@
   // -----
 
   var $truthy = Opal.truthy = function(val) {
-    return false !== val && nil !== val && undefined !== val && null !== val && (!(val instanceof Boolean) || true === val.valueOf());
```

Comment:
There are two specs that are failing in which they explicitly use a boxed boolean, not sure if anything changed so that it's a super contrived example and basically impossible to recreate that situation, but we need to be sure that's the case before dropping support for this.

Reply:
I have worked together with @merceroncode and we noted that just simply adding `return true` improved performance slightly.

I have worked with you @elia together in the past on this line of code. I am in support of dropping support for boxed booleans for a couple of reasons:

* They don't occur naturally in Opal code
* They aren't even supported by JavaScript

Reply:
🙌 Happy to drop support given it's all expected to be strict code now:

```rb
`typeof #{true.itself}` # => 'boolean'
```